hack/kind-up.sh

```diff
@@ -53,6 +53,27 @@ parse_flags() {
   done
 }
 
+check_local_dns_records() {
+  local glgc_ip_address
+  glgc_ip_address=""
+
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    # Suppress exit code using "|| true"
+    glgc_ip_address=$(dscacheutil -q host -a name garden.local.gardener.cloud | grep "ip_address" | head -n 1| cut -d' ' -f2 || true)
+  elif [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    # Suppress exit code using "|| true"
+    glgc_ip_address="$(getent hosts garden.local.gardener.cloud | cut -d' ' -f1 || true)"
+  else
+    echo "Warning: Unknown OS. Make sure garden.local.gardener.cloud resolves to 127.0.0.1"
+    return 0
+  fi
+
+  if [ "$glgc_ip_address" != "127.0.0.1" ]; then
+      echo "Error: garden.local.gardener.cloud does not resolve to 127.0.0.1. Please add a line for it in /etc/hosts"
```

Comment:
If the e2e tests should run in prow, you have to add this resolution to `/etc/hosts`.
We are doing something similar for the DNS names of the shoots [here](https://github.com/gardener/gardener/blob/00b41a217ff62f25842e4ed53a0b99b319ecec84/hack/test-e2e-local.sh#L105-L119). The existence of `CI` and `ARTIFACTS` envs can tell you that the test is running in prow.

Reply:
Thank you! I haven't had the time to look into it, so you just saved me some time. I'll finish this up soon